src/App.tsx

```diff
@@ -1,5 +1,45 @@
+import { useEffect, useState } from 'react'
+import SpotifyWebApi from 'spotify-web-api-js'
+import Login from './components/Login'
+import MainPage, { UserData } from './components/MainPage/MainPage'
+import { getTokenFromUrl } from './helper/spotify'
+
 function App() {
-  return <div>Hello World</div>
+  const fallbackUserData: UserData = {
+    country: 'DE',
+    display_name: 'John Doe',
+    product: 'premium',
+  }
+
+  const [token, setToken] = useState<string | null>(null)
+  const [userData, setUserData] = useState<UserData>(fallbackUserData)
+
+  const spotify = new SpotifyWebApi()
+
+  useEffect(() => {
+    setToken(getTokenFromUrl())
+  }, [])
+
+  useEffect(() => {
+    if (token) {
+      window.location.hash = ''
+      spotify.setAccessToken(token)
+      fetchUserData()
+    }
+  }, [token])
+
+  return <div>{token ? <MainPage userData={userData} /> : <Login></Login>}</div>
```

Comment:
I am not sure about the `fallbackUserData`. When is this important?

Without it, you could reduce the function even more:

```suggestion
  const [userData, setUserData] = useState<UserData | null>(null)

  const spotify = new SpotifyWebApi()

  useEffect(() => {
    const token = getTokenFromUrl();
    if (token) {
      window.location.hash = ''
      spotify.setAccessToken(token)
      fetchUserData()
    }
  }, [])

  return <div>{userData ? <MainPage userData={userData} /> : <Login></Login>}</div>
```